src/option.c

```diff
@@ -984,11 +1005,35 @@ set_init_3(void)
      * Set 'shellcmdflag', 'shellxquote', and 'shellquote' depending on the
      * 'shell' option.
      * This is done after other initializations, where 'shell' might have been
-     * set, but only if they have not been set before.  Default for p_shcf is
-     * "/c", for p_shq is "".  For "sh" like  shells it is changed here to
-     * "-c" and "\"".  And for Win32 we need to set p_sxq instead.
+     * set, but only if they have not been set before.
+     * Default values depend on shell (cmd.exe is default shell):
+     *
+     *			    p_shcf	p_sxq
+     * cmd.exe          -   "/c"	"("
+     * powershell.exe   -   "-Command"	"\""
+     * "sh" like shells -   "-c"	"\""
+     *
+     * For Win32 p_sxq is set instead of p_shq to include shell redirection.
      */
-    if (strstr((char *)gettail(p_sh), "sh") != NULL)
+    if (strstr((char *)gettail(p_sh), "powershell") != NULL)
+    {
+	int	idx_opt;
+
+	idx_opt = findoption((char_u *)"shcf");
+	if (idx_opt >= 0 && !(options[idx_opt].flags & P_WAS_SET))
+	{
+	    p_shcf = (char_u*)"-Command";
+	    options[idx_opt].def_val[VI_DEFAULT] = p_shcf;
+	}
+
+	idx_opt = findoption((char_u *)"sxq");
+	if (idx_opt >= 0 && !(options[idx_opt].flags & P_WAS_SET))
+	{
+	    p_sxq = (char_u*)"\"";
+	    options[idx_opt].def_val[VI_DEFAULT] = p_sxq;
+	}
+    }
+    else if (strstr((char *)gettail(p_sh), "sh") != NULL)
```

Comment:
Can you add some tests for setting 'shell' to 'powershell'?

Reply:
Will do. Not done much before so will need to get up to speed. Should this include running a powershell shell as well? Is such an environment available?

Reply:
You can set 'shell' to 'powershell' and then check the value of the
various modified options. You can then reset 'shell' to the default
value and then check the value of the options again. I think this
can be done even if powershell is not present in the environment.

You can also add tests specific to MS-Windows (assuming powershell
is always present or you can check powershell is present in PATH and
is executable), where you can set 'shell' to powershell and
run some simple external commands and check for the quotes.

Reply:
I have added shell related tests in #8464. This can be extended to check for powershell.

Reply:
Thanks for that, it would have taken me a while to get to that state.  I can learn from this. And why is the popup filter test failing? It seems to vary by build and run.

Reply:
> why is the popup filter test failing?
here: https://github.com/vim/vim/pull/8409#issuecomment-864609993

Reply:
>
> I have added shell related tests in #8464. This can be extended to check for powershell.
>

This PR has been merged now (8.2.3061). With your changes, you need to update
this test to account for the new shell settings when using PowerShell.